rabit/include/rabit/internal/socket.h

```diff
@@ -21,31 +24,92 @@
 #include <netinet/in.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
+
 #endif  // defined(_WIN32)
+
 #include <string>
 #include <cstring>
 #include <vector>
 #include <unordered_map>
 #include "utils.h"
 
-#if defined(_WIN32) || defined(__MINGW32__)
+#if defined(_WIN32) && !defined(__MINGW32__)
 typedef int ssize_t;
 #endif  // defined(_WIN32) || defined(__MINGW32__)
 
 #if defined(_WIN32)
-typedef int sock_size_t;
+using sock_size_t = int;
 
-static inline int poll(struct pollfd *pfd, int nfds,
-                       int timeout) { return WSAPoll ( pfd, nfds, timeout ); }
 #else
+
 #include <sys/poll.h>
 using SOCKET = int;
 using sock_size_t = size_t;  // NOLINT
-const int kInvalidSocket = -1;
 #endif  // defined(_WIN32)
 
+#define IS_MINGW() defined(__MINGW32__)
```

Comment:
Ifdef'ing everything seems quite a messy solution. Any alternatives?

Reply:
I don't have any other options.

Reply:
I can think of anything better, as you said at least it's constrained to one file.